### PR TITLE
 refactor(validateUrlEncoding)

### PR DESCRIPTION
**Pull request by M4tteoP**
Stumbled into some code that deserved some care.
- Improved Coverage (there were no tests)
- Revisited the code
- FWIW, it also brings some performance improvements to that operator

```
name                    old time/op    new time/op    delta
ValidateURLEncoding-10    29.1ns ± 2%    20.5ns ± 3%  -29.52%  (p=0.000 n=19+18)
name                    old alloc/op   new alloc/op   delta
ValidateURLEncoding-10     0.00B          0.00B          ~     (all equal)
```

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved URL-encoding validation: empty inputs are now treated consistently and percent-encoding issues (non-hex digits or truncated sequences) are detected more reliably.

* **Tests**
  * Added unit tests covering URL-encoding edge cases and a benchmark to track performance and regression over time.

[Review Change Stack](https://app.coderabbit.ai/change-stack/corazawaf/coraza/pull/1626?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by fzipi — COMMENTED** [review]
Ugh... I hate doing this, but can't we just use an enum with proper names for the result instead of 1/-2/-3?

**Comment by M4tteoP** [comment]
Makes sense, I’ll update it. I honestly didn’t pay much attention to what we were returning, considering we’re not using it, and it’s just returning the bool anyway.

File: internal/operators/validate_url_encoding.go [hunk 1]
```diff
@@ -9,6 +9,14 @@ import (
 	"github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
 )
 
+type urlEncodingResult int
+
+const (
+	urlEncodingValid urlEncodingResult = iota
+	urlEncodingInvalidNonHex
+	urlEncodingInvalidTruncated
```
[the remark this answers is not in the document]
Comment (M4tteoP) [reply internal/operators/validate_url_encoding.go:17]: Tell me more about this guideline compliance

**Comment by M4tteoP** [comment]
Done @fzipi, ready for a second pass. thx!

**Review by fzipi — APPROVED** [review]
🫶🫶🫶
